### PR TITLE
fix: avoid grouping logs from tasks which ended in error

### PR DESCRIPTION
**Pull request by chris-olszewski**
### Description

Fixes https://github.com/vercel/turbo/issues/6709

Github Action doesn't provide a mechanism for grouping logs together and keeping them expanded which is helpful for highlighting errors.

This PR changes the behavior so we avoid emitting the grouping header/footer if task execution was a failure.

### Testing Instructions

Updated GH Action log grouping integration test

Closes TURBO-2320

### CODE REVIEW

**Review by mehulkar — COMMENTED** [review]
This is great. Another option, which I could see myself preferring is to keep the grouping behavior, but make the header and footer red on failure. What do you think about that?

**Comment by chris-olszewski** [comment]
> make the header and footer red on failure

Like [this](https://github.com/vercel/turbo/actions/runs/7851566660/job/21428813279?pr=7341#step:3:6)?

**Comment by mehulkar** [comment]
Yeah exactly. If the arrow could be red too that would be great. I messed around in dev tools to get this: 
CleanShot 2024-02-10 at 12 36 36@2x

This is using the same red as we get at the end, but I actually don't think it's dark enough. Ideally, I want to be able to scroll through a long list of collapsed groups and visually find the one that failed. Maybe a big ❌ in the task prefix would do the trick (but not sure that's technically feasible since the prefix is computed ahead of time)
CleanShot 2024-02-10 at 12 38 41@2x

File: crates/turborepo-lib/src/task_graph/visitor.rs [hunk 1]
```diff
@@ -690,7 +690,10 @@ impl ExecContext {
             .instrument(span)
             .await;
 
-        let logs = match output_client.finish() {
+        // If the task resulted in an error, do not group in order to better highlight
+        // the error.
+        let keep_group = matches!(result, ExecOutcome::Success(_));
```
Comment (tknickman) [inline crates/turborepo-lib/src/task_graph/visitor.rs:695]: Woah, TIL about `matches!`. Love that.

**Review by tknickman — APPROVED** [review]

**Pull request merged**